verifier/truststore/errors.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package truststore
+
+// TrustStoreError is used when accessing specified trust store failed
+type TrustStoreError struct {
```

Comment:
We have two errors of similar time and I do see need of this kind of errors in future were we want user friendly error message but still want to preserve orignal error. Should we create an generic WrappedError struct?

Reply:
@priteshbandi Not sure if I understand your comment, what do you mean by `two errors of similar time`? 
The original error is preserved. For example when `TrustStoreError` is created, it would be `TrustStoreError{InnerError: fmt.Errorf("failed to access the trust store %q: %w", path, err)}` , note, the original error is wrapped. If the caller wants the inner error, they can just call `TrustStoreError`'s `Unwrap()` method.

Reply:
Oh, and one reason to separate `TrustStoreError` and `CertificateError` is that they might have different actionable next steps. So the caller such as Notation CLI needs a way to tell apart these two types of errors.